rules/rulelist.go

```diff
@@ -0,0 +1,93 @@
+// (c) Copyright 2016 Hewlett Packard Enterprise Development LP
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"github.com/GoASTScanner/gas"
+)
+
+type RuleDefinition struct {
+	Description string
+	Create      gas.RuleBuilder
+}
+
+type RuleList map[string]RuleDefinition
+
+func (rl RuleList) Builders() []gas.RuleBuilder {
+	builders := make([]gas.RuleBuilder, 0, len(rl))
+	for _, def := range rl {
+		builders = append(builders, def.Create)
+	}
+	return builders
+}
+
+type RuleFilter func(string) bool
+
+func NewRuleFilter(action bool, ruleIDs ...string) RuleFilter {
+	rulelist := make(map[string]bool)
+	for _, rule := range ruleIDs {
```

Comment:
I would validate if the ruleID is really defined in the list. If not, I will skip it.